WeakAuras/Modernize.lua

```diff
@@ -2013,6 +2013,69 @@ function Private.Modernize(data)
 
   end
 
+  -- Version 72 moves instance filters from a conditions trigger to a player location trigger
+  if data.internalVersion < 72 then
+    local new_triggers = {}
+    local delete_triggers = {}
+    local condition_filters = {
+      "alwaystrue",
+      "incombat",
+      "pvpflagged",
+      "alive",
+      "vehicle",
+      "resting",
+      "mounted",
+      "HasPet",
+      "ismoving",
+      "afk",
+      "ingroup",
+    }
+
+    for i, triggerData in ipairs(data.triggers) do
+      local t = triggerData.trigger
+      local canRemoveTrigger = true
+
+      -- Check for a Conditions trigger with instance filters
+      if t.event == "Conditions" and (
+        t.use_instance_size
+        or t.use_instance_difficulty
+        or t.use_instance_type
+      ) then -- Create a new trigger object
+        local new = CopyTable(triggerData)
+        new.trigger.event = "Player Location"
+        new.trigger.use_instanceDifficulty = new.trigger.use_instance_difficulty
+        new.trigger.use_instance_difficulty = nil
+        new.trigger.instanceDifficulty = new.trigger.instance_difficulty
+        new.trigger.instance_difficulty = nil
+
+        -- Remove any condition filters from the new trigger
+        for _, name in ipairs(condition_filters) do
+          if t["use_"..name] ~= nil then
+            canRemoveTrigger = false
+          end
+          new.trigger[name] = nil
+          new.trigger["use_"..name] = nil
+        end
+        tinsert(new_triggers, new)
+      end
+
+      -- the remaining conditions trigger has no active filters and the trigger can be removed
+      if canRemoveTrigger then
+        delete_triggers[i] = true
+      end
+    end
+
+    -- Remove triggers
+    for i in pairs(delete_triggers) do
+      data.triggers[i] = nil
+    end
+
+    -- Add triggers
+    for _,triggerData in ipairs(new_triggers) do
+      tinsert(data.triggers, triggerData)
+    end
+  end
```

Comment:
This migration ignore conditions, display texts, and trigger activation (i don't fathom how custom could be handled)

Reply:
ah I suspected it wouldn't be that simple. I'll look into it.
I was already leaning a bit to just keeping instance filters untouched, so this might be a good reason to just leave it

Reply:
Since a new expansion (for both classic & mainline) is on the horizon, another option is to instead add an AuraWarning for auras that use conditions, and then drop support for them on the expansion.